src/frontend/src/optimizer/plan_node/logical_window_agg.rs

```diff
@@ -0,0 +1,290 @@
+    PredicatePushdown, ToBatch, ToStream,
+};
+use crate::expr::{Expr, ExprImpl, InputRef, InputRefDisplay, WindowFunction, WindowFunctionType};
+use crate::utils::{ColIndexMapping, Condition};
+
+/// Rewritten version of [`WindowFunction`] which uses `InputRef` instead of `ExprImpl`.
+#[derive(Debug, Clone)]
+pub struct PlanWindowFunction {
+    pub function_type: WindowFunctionType,
+    pub return_type: DataType,
+    pub partition_by: Vec<InputRef>,
+    /// TODO: rename & move `PlanAggOrderByField` so that it can be better shared like
+    /// [`crate::expr::OrderByExpr`]
+    pub order_by: Vec<PlanAggOrderByField>,
+}
+
+struct PlanWindowFunctionDisplay<'a> {
+    pub window_function: &'a PlanWindowFunction,
+    pub input_schema: &'a Schema,
+}
+
+impl<'a> std::fmt::Debug for PlanWindowFunctionDisplay<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let window_function = self.window_function;
+        if f.alternate() {
+            f.debug_struct("WindowFunction")
+                .field("function_type", &window_function.function_type)
+                .field("return_type", &window_function.return_type)
+                .field("partition_by", &window_function.partition_by)
+                .field("order_by", &window_function.order_by)
+                .finish()
+        } else {
+            write!(f, "{}() OVER(", window_function.function_type.name())?;
+
+            let mut delim = "";
+            if !window_function.partition_by.is_empty() {
+                delim = " ";
+                write!(
+                    f,
+                    "PARTITION BY {}",
+                    window_function
+                        .partition_by
+                        .iter()
+                        .format_with(", ", |input_ref, f| {
+                            f(&InputRefDisplay {
+                                input_ref,
+                                input_schema: self.input_schema,
+                            })
+                        })
+                )?;
+            }
+            if !window_function.order_by.is_empty() {
+                write!(
+                    f,
+                    "{delim}ORDER BY {}",
+                    window_function.order_by.iter().format_with(", ", |e, f| {
+                        f(&format_args!(
+                            "{:?}",
+                            PlanAggOrderByFieldDisplay {
+                                plan_agg_order_by_field: e,
+                                input_schema: self.input_schema,
+                            }
+                        ))
+                    })
+                )?;
+            }
+            f.write_str(")")?;
+
+            Ok(())
+        }
+    }
+}
+
+/// `LogicalWindowAgg` performs `OVER` window aggregates ([`WindowFunction`]) to its input.
+///
+/// The output schema is the input schema plus the window functions.
+#[derive(Debug, Clone)]
+pub struct LogicalWindowAgg {
```

Comment:
```suggestion
pub struct LogicalWindow {
```
actually, here the operator does not do the "aggregation". so I think we should not use "Agg" here.
Also, I think the stream window operators are special cases,  so they should be operators like "HopWindow"